bigquery-storage/api/BigQueryStorage.Samples.Tests/BigQueryStorageFixture.cs

```diff
@@ -0,0 +1,98 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Google.Api.Gax;
+using Google.Apis.Bigquery.v2.Data;
+using Google.Cloud.BigQuery.V2;
+using Grpc.Core;
+using Xunit;
+
+[CollectionDefinition(nameof(BigQueryStorageFixture))]
+public class BigQueryStorageFixture : IDisposable, ICollectionFixture<BigQueryStorageFixture>
+{
+    private BigQueryClient _client;
+
+    internal List<string> TempDatasetIds { get; } = new List<string>();
+
+    internal string ProjectId { get; }
+
+    public BigQueryStorageFixture()
+    {
+        ProjectId = Environment.GetEnvironmentVariable("GOOGLE_PROJECT_ID");
+        if (string.IsNullOrWhiteSpace(ProjectId))
+        {
+            throw new Exception("Missing GOOGLE_PROJECT_ID. You need to set the Environment variable 'GOOGLE_PROJECT_ID' with your Google Cloud Project's project id.");
+        }
+
+        _client = BigQueryClient.Create(ProjectId);
+    }
+
+    /// <summary>
+    /// Ensures the dataset and table with given name and schema exists
+    /// </summary>
+    /// <param name="datasetId">The BigQuery Dataset Id</param>
+    /// <param name="tableName">The BigQuery table name</param>
+    /// <param name="tableSchema">The schema for table</param>
+    internal void EnsureDataSetAndTableExists(string datasetId, string tableName, TableSchema tableSchema)
+    {        
+        BigQueryDataset dataSet = _client.GetOrCreateDataset(datasetId);
+        if (!TempDatasetIds.Contains(datasetId))
+        {
+            TempDatasetIds.Add(datasetId);
+        }
+
+        dataSet.GetOrCreateTable(tableName, tableSchema);
+    }
+
+    /// <summary>
+    ///  Disposes the instance of BigQueryStorageFixture
+    /// </summary>
+    public void Dispose()
+    {
+        try
+        {
+            foreach (string datasetId in TempDatasetIds)
+            {
+                DeleteDataSet(datasetId);
+            }            
+        }
+        catch (RpcException)
+        {
+            // Do nothing, we are deleting on a best effort basis.
+        }
+    }
+
+    /// <summary>
+    ///  Gets the table data rows for the given table name in the BigQuery Dataset.
+    /// </summary>
+    /// <returns></returns>
+    internal async Task<PagedEnumerable<TableDataList, BigQueryRow>> GetTableDataAsync(string datasetId, string tableName)
+    {
+        BigQueryTable table = await _client.GetTableAsync(datasetId, tableName);
+        return table.ListRows();
+    }
+
+    /// <summary>
+    /// Deletes the dataset 
+    /// </summary>
+    internal void DeleteDataSet(string datasetId)
+    {
+        _client.DeleteDataset(datasetId, new DeleteDatasetOptions() { DeleteContents = true });
```

Comment:
Nits just to get into conventional habits:

- I'd remove the `()` here
- The method name should have a lowercase 's' in 'Dataset'
- I'd use an expression-bodied member

So:

```csharp
internal void DeleteDataset(string datasetId) =>
    _client.DeleteDataset(datasetId, new DeleteDatasetOptions { DeleteContents = true });
```

Reply:
Thanks Jon. Noted. Done